packages/core/src/provider.tsx

```diff
@@ -52,7 +52,12 @@ export function Web3ReactProvider<T extends BaseProvider = Web3Provider>({
   const error = usePriorityError()
   const account = usePriorityAccount()
   const isActive = usePriorityIsActive()
-  const provider = usePriorityProvider<T>(network)
+  // note that we've omitted a <T extends BaseProvider = Web3Provider> generic type
+  // in Web3ReactProvider, and thus can't pass T through to usePriorityProvider below.
+  // this is because if we did so, the type of provider would include T, but that would
+  // conflict because Web3Context can't take a generic. as a work-around,
```

Comment:
idk that this is a "work around", as much as a more correct way of defining types here.